distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/common/downloads/InterruptableAsyncCharConsumer.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.test.itests.common.downloads;
+
+import java.io.IOException;
+import java.nio.CharBuffer;
+
+import org.apache.http.HttpException;
+import org.apache.http.HttpResponse;
+import org.apache.http.nio.IOControl;
+import org.apache.http.nio.client.methods.AsyncCharConsumer;
+import org.apache.http.protocol.HttpContext;
+
+/**
+ * Socket ingest strategy for testing the download of products.
+ */
+public class InterruptableAsyncCharConsumer extends AsyncCharConsumer<String> {
+    private StringBuilder builder = new StringBuilder();
```

Comment:
If we don't want to cache the results from the closed client using some form of consumer in `org.apache.http` then this can be removed.